selene/src/main.rs

```diff
@@ -412,15 +412,28 @@ fn start(mut options: opts::Options) {
 
                 (config_contents, Path::new("-"))
             } else {
-                let config_path = Path::new("selene.toml");
-
-                let config_contents = match fs::read_to_string(config_path) {
-                    Ok(contents) => contents,
-                    Err(error) => {
-                        error!("Error reading config file: {error}");
-                        std::process::exit(1);
+                let config_paths_to_check = [Path::new("selene.toml"), Path::new(".selene.toml")];
+                let mut config_contents = String::new();
+                let mut config_path = Path::new("");
+                let mut config_not_found = true;
+
+                for path in &config_paths_to_check {
+                    match fs::read_to_string(path) {
+                        Ok(contents) => {
+                            config_contents = contents;
+                            config_path = path;
+                            config_not_found = false;
+                            break;
+                        }
+                        Err(error) => {
+                            error!("Error reading config file: {error}");
+                        }
                     }

```

Comment:
This wouldn't work because you're propagating `fs::read_to_string` errors even when the path doesn't exist. You'd need to handle that separately:

```suggestion
                [".selene.toml", "selene.toml"]
                    .iter()
                    .find_map(|path| {
                        let path = Path::new(path);
                        path.exists().then_some(path).and_then(|path| {
                            fs::read_to_string(path).map_or_else(
                                |e| {
                                    error!("Error reading config file: {e}");
                                    std::process::exit(1);
                                },
                                |contents| Some((contents, path)),
                            )
                        })
                    })
                    .unwrap_or_else(|| {
                        error!("No config file found");
                        std::process::exit(1);
                    })
```

Reply:
fixed in https://github.com/Zeioth/selene/commit/8947dea186d45ba65b5e8196b670bb6750ef83b8